docs/faq.md

```diff
@@ -63,3 +63,29 @@ linear = eqx.nn.Linear(input_size, output_size, key=key)
 
 y = jax.vmap(linear)(x)
 ```
+
+## TypeError: not a valid JAX type.
+
+You might be getting an error like
+```
+TypeError: Argument '<function ...>' of type <class 'function'> is not a valid JAX type.
+```
+Example:
+```python3
+import jax
+import equinox as eqx
+
+def loss_fn(model, x, y):
+    return (model(x) - y) ** 2
+
+model = eqx.nn.Lambda(lambda x: x)
```

Comment:
I suspect `Lambda` and `MLP` are going to be the two most frequent culprits here. Can we add both to the example here? E.g:
```python
# Use a model that includes a non-array in its pytree structure
model = eqx.nn.Lambda(jax.nn.relu)
model = eqx.nn.MLP(2, 2, 2, 2, key=jax.random.PRNGKey(0))
```
(or similar)